cmd/cli/version_test.go

```diff
@@ -0,0 +1,188 @@
+			namespace: "test",
+			remoteVersion: &version.Info{
+				Version:   "v0.0.0",
+				GitCommit: "xxxxxxx",
+				BuildDate: "Date",
+			},
+			remoteVersionInfo: &remoteVersionInfo{
+				meshName: "osm",
+				version: &version.Info{
+					Version:   "v0.0.0",
+					GitCommit: "xxxxxxx",
+					BuildDate: "Date",
+				},
+			},
+			controllerPods: []*corev1.Pod{
+				{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "controllerPod",
+						Namespace: "test",
+						Labels: map[string]string{
+							constants.AppLabel:               constants.OSMControllerName,
+							constants.OSMAppInstanceLabelKey: "osm",
+						},
+					},
+				},
+			},
+			err: nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			assert := tassert.New(t)
+
+			buf := bytes.NewBuffer(nil)
+
+			objs := make([]runtime.Object, len(test.controllerPods))
+			for i := range test.controllerPods {
+				objs[i] = test.controllerPods[i]
+			}
+
+			cmd := versionCmd{
+				out:       buf,
+				namespace: test.namespace,
+				clientset: fake.NewSimpleClientset(objs...),
+				remoteVersion: &fakeRemoteVersion{
+					err:         test.err,
+					versionInfo: test.remoteVersion,
+				},
+				remote: true,
+			}
+
+			remoteVersionInfo, err := cmd.getMeshVersion()
+			if err != nil && test.err != nil {
+				assert.Equal(test.err.Error(), err.Error())
+			} else {
+				assert.Equal(test.err, err)
+			}
+
+			assert.Equal(test.remoteVersionInfo, remoteVersionInfo)
+		})
+	}
+}
+
+func TestOutputVersionInfo(t *testing.T) {
+	tests := []struct {
+		name        string
+		versionInfo versionInfo
+		expected    string
+		remote      bool
+	}{
+		{
+			name: "cli and no mesh",
+			versionInfo: versionInfo{
+				cliVersionInfo: &version.Info{
+					Version:   "v0.0.0",
+					GitCommit: "xxxxxxx",
+					BuildDate: "0000-00-00-00:00",
+				},
+			},
+			expected: "CLI Version: version.Info{Version:\"v0.0.0\", GitCommit:\"xxxxxxx\", BuildDate:\"0000-00-00-00:00\"}\n",
+			remote:   false,
+		},
+		{
+			name: "cli and mesh",
+			versionInfo: versionInfo{
+				cliVersionInfo: &version.Info{
+					Version:   "v0.0.0",
+					GitCommit: "xxxxxxx",
+					BuildDate: "0000-00-00-00:00",
+				},
+				remoteVersionInfo: &remoteVersionInfo{
+					meshName: "test",
+					version: &version.Info{
+						Version:   "v0.0.0",
+						GitCommit: "xxxxxxx",
+						BuildDate: "0000-00-00-00:00",
+					},
+				},
+			},
+			expected: "CLI Version: version.Info{Version:\"v0.0.0\", GitCommit:\"xxxxxxx\", BuildDate:\"0000-00-00-00:00\"}\nMesh [test] Version: version.Info{Version:\"v0.0.0\", GitCommit:\"xxxxxxx\", BuildDate:\"0000-00-00-00:00\"}\n",
+			remote:   true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			assert := tassert.New(t)
+
+			buf := bytes.NewBuffer(nil)
+			cmd := versionCmd{
+				out:    buf,
+				remote: test.remote,
+			}
+			cmd.outputVersionInfo(test.versionInfo)
+
+			assert.Contains(buf.String(), test.expected)
```

Comment:
`Equal` also passes here, so I'd use that just to avoid any mistakes where e.g. we expect only the client version but both the client and remote versions get written.
```suggestion
			assert.Equal(buf.String(), test.expected)
```